prompt2model/utils/api_tools.py

```diff
@@ -73,6 +80,7 @@ def generate_one_completion(
             An OpenAI-like response object if there were no errors in generation.
             In case of API-specific error, Exception object is captured and returned.
         """
+        max_tokens = self.max_tokens or 4 * count_tokens_from_string(prompt)
```

Comment:
I think that `max_tokens` should be scaled back to the longest context length of the agent (for example, 4k). So, it's actually hard to set this parameter.

Reply:
At least, if the `max_tokens` is too long, this API call will fail. 🤔

Reply:
Fixed: https://github.com/neulab/prompt2model/pull/344/commits/4424a05a756b4174e6a9f3acf3c8f05331c13939